### PR TITLE
Add CLI support through `python -mbracex`

### LINKED ISSUE #27: Feature proposal/offer: command line interface
I've been [seeking a command `x`, like `seq` but doing brace expansion](https://stackoverflow.com/questions/69383322/is-there-command-like-seq-that-performs-full-bash-like-brace-expansion-to-std). It could be used like

```bash
x "{a..z}/{a..z}/{a..z}/{a..z}/{a..z}/{a..z}" | xargs -d $'\n' mkdir -p
```

I've not found a satisfactory `x', so I'd like to offer a PR allowing `python -m bracex` to serve as `x`. Would you be interested?

I imagine the major elements would be
- a new `__main__.py` wrapping `bracex.iexpand()`
- minimal argument parsing using `argparse`
- updates to documentation
- additional tests to maintain code coverage

I totally understand if the answer is no, so please don't feel pressured. I know every new feature creates support costs. As plan B I could release a separate package that wraps bracex, keeping a loose coupling.

Thanks for your consideration, Alex

[1]:

### PR DESCRIPTION
**Pull request by moreati**
fixes #27

TODO
- [x] Changelog
- [x] Performance test
- [x] Consider parametrised unit tests
- [x] Consider a (default) strict mode (e.g. should an expression with no expansions be considered an error?)

### CODE REVIEW

**Comment by moreati** [comment]
I dropped `--separator` in favour of `--terminator` and `-0` because
- having a terminator is more consistent with the behaviour of `find -print` or `find -print0`
- It's very hard or not possible to pass a NUL character as a string argument in bash.

Further discussion/background https://twitter.com/moreati/status/1443321003606745103

File: docs/src/dictionary/en-custom.txt [hunk 1]
```diff
@@ -8,24 +8,29 @@ Gitter
...
 Virtualenv
+Willmer
 accessor
 deprecations
 expander
 globbing
 incrementer
 iterables
 linter
+nocover
```
Comment (facelessuser) [inline docs/src/dictionary/en-custom.txt:26]: These are normally needed. They should be ignored automatically due to the PySpelling config.

Comment (facelessuser) [reply docs/src/dictionary/en-custom.txt:26]: To clarify, both `nocover` and `pragma` will be ignored.

Comment (facelessuser) [reply docs/src/dictionary/en-custom.txt:26]:
```console
➜  bracex git:(master) python3 -m pyspelling -n python
Spelling check passed :)
```

Notice I already have `# pragma: no cover` in source: https://github.com/facelessuser/bracex/blob/master/bracex/__meta__.py#L134

Comment (facelessuser) [reply docs/src/dictionary/en-custom.txt:26]: Use `# pragma: no cover`. That's why it failed.

File: tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+"""Test command module and argument handling."""
+import pytest
+
```
Comment (facelessuser) [inline tests/test_main.py:3]: Please remove the extra new line.

File: docs/src/markdown/about/changelog.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## 2.2
 
 - **NEW** Support Python 3.10
+- **NEW** Command line interface using `python3 -m bracex`
```
Comment (facelessuser) [inline docs/src/markdown/about/changelog.md:6]: I just noticed this. To be consistent, can we add the colon after: `**NEW**:`?

File: bracex/__main__.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import argparse
+
+import bracex
+
+
+def main(argv=None):
+    """Accept command line arguments and output brace expansion to stdout."""
+    parser = argparse.ArgumentParser(
+        prog='python -mbracex',
+        description=__doc__.lstrip('\n').split('\n')[0],
```
Comment (facelessuser) [inline bracex/__main__.py:33]: While clever, I'd prefer we just copy the string here. It's a lot of work to get a single short string 🙂. It'll also be easier for people to understand in general.

**Review by facelessuser — CHANGES_REQUESTED** [review]
Overall, it looks good! I've added a few comments on things that should be addressed.

File: README.md [hunk 5]
```diff
@@ -44,6 +60,7 @@ Documentation is found here: https://facelessuser.github.io/bracex/.
 MIT License
 
 Copyright (c) 2018 - 2021 Isaac Muse
+Copyright (c) 2021 Alex Willmer
```
[the remark this answers is not in the document]
Comment (facelessuser) [reply README.md:63]: Yeah, it caught me off guard, but no worries 🙂.

**Comment by facelessuser** [comment]
I'll probably give it a little spin in the next few days and release this weekend.

**Pull request merged**